prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -431,11 +438,18 @@ class ReportFunction(
         if (sender == null)
             actionLogs.error(InvalidParamMessage("'$CLIENT_PARAMETER:$clientName': unknown sender"))
 
-        val schema = workflowEngine.metadata.findSchema(sender?.schemaName ?: "")
-        if (sender != null && schema == null)
-            actionLogs.error(
-                InvalidParamMessage("'$CLIENT_PARAMETER:$clientName': unknown schema '${sender.schemaName}'")
-            )
+        // verify schema if the sender is a covidSender
+        var missingRequiredSchema = false
+        var schema: Schema? = null
+        if (sender != null && sender is CovidSender) {
+            schema = workflowEngine.metadata.findSchema(sender.schemaName)
+            if (schema == null) {
+                missingRequiredSchema = true
```

Comment:
This variable can go since the if in 466 tests actionLogs.hasErrors

Reply:
Updated.